Support (or don't crash) for unsupported SGR code

See https://github.com/bpython/bpython/issues/657

    mic@mic ~ $ bpython3
    bpython version 0.16 on top of Python 3.5.2+ /usr/bin/python3
    >>> print("\x1b[2ma")
    a
    Traceback (most recent call last):
      File "/usr/bin/bpython3", line 11, in <module>
        load_entry_point('bpython==0.16', 'console_scripts', 'bpython')()
      File "/usr/lib/python3/dist-packages/bpython/curtsies.py", line 190, in main
        exit_value = repl.mainloop()
      File "/usr/lib/python3/dist-packages/bpython/curtsies.py", line 125, in mainloop
        self.process_event_and_paint(e)
      File "/usr/lib/python3/dist-packages/bpython/curtsies.py", line 100, in process_event_and_paint
        scrolled = self.window.render_to_terminal(array, cursor_pos)
      File "/usr/lib/python3/dist-packages/curtsies/window.py", line 409, in render_to_terminal
        if line == self._last_lines_by_row.get(row, None):
      File "/usr/lib/python3/dist-packages/curtsies/formatstring.py", line 367, in __eq__
        return str(self) == str(other)
      File "/usr/lib/python3/dist-packages/curtsies/formatstring.py", line 328, in __unicode__
        self._unicode = ''.join(unicode(fs) for fs in self.basefmtstrs)
      File "/usr/lib/python3/dist-packages/curtsies/formatstring.py", line 328, in <genexpr>
        self._unicode = ''.join(unicode(fs) for fs in self.basefmtstrs)
      File "/usr/lib/python3/dist-packages/curtsies/formatstring.py", line 106, in __unicode__
        value = self.color_str
      File "/usr/lib/python3/dist-packages/curtsies/formatstring.py", line 96, in color_str
        assert k in xforms, "XXX Do we actually get cases like this?"
    AssertionError: XXX Do we actually get cases like this?

--- a/curtsies/formatstring.py
+++ b/curtsies/formatstring.py
@@ -93,8 +93,10 @@ class Chunk(object):
         s = self.s
         for k, v in sorted(self.atts.items()):
             # (self.atts sorted for the sake of always acting the same.)
-            assert k in xforms, "XXX Do we actually get cases like this?"
-            if v is False:
+            if k not in xforms:
+                # Unsupported SGR code
+                continue
+            elif v is False:
                 continue
             elif v is True:
                 s = xforms[k](s)

--- a/tests/test_fmtstr.py
+++ b/tests/test_fmtstr.py
@@ -59,6 +59,8 @@ class TestFmtStrParsing(unittest.TestCase):
         self.assertEqual(str(fmtstr('\x1b[43mhello\x1b[0m')), '\x1b[43mhello\x1b[49m')
         self.assertEqual(str(fmtstr('\x1b[43mhello')), '\x1b[43mhello\x1b[49m')
         self.assertEqual(str(fmtstr('\x1b[32;1mhello')), '\x1b[32m\x1b[1mhello\x1b[0m\x1b[39m')
+        self.assertEqual(str(fmtstr('\x1b[2mhello')), 'hello')
+        self.assertEqual(str(fmtstr('\x1b[32;2mhello')), '\x1b[32mhello\x1b[39m')
         self.assertEqual(str(fmtstr('\x1b[33m\x1b[43mhello\x1b[0m')),
                          '\x1b[33m\x1b[43mhello\x1b[49m\x1b[39m')
 

